enigmampc/integration-tests
# Refactor tests

- Refactor tests code
  - Reorg tests so they'll be more self-contained instead of relying on previous tests and shared files :heavy_check_mark: 
  - Refactor duplicate code into util functions :heavy_check_mark: 
  - Merge new tests @AvishaiW wrote :heavy_check_mark: 
  - Use Prettier :heavy_check_mark: 
- Refactor CI:
  - Split steps for clarity :heavy_check_mark: 
  - Add caching for better performance (docker, node_modules, etc.)
  - Check if we can run privately in HW mode :heavy_check_mark:
